pkg/storaged/client.js

```diff
@@ -422,7 +422,11 @@ function init_model(callback) {
     }
 
     function enable_nfs_features() {
-        return cockpit.spawn([ "which", "mount.nfs" ], { err: "ignore" }).then(
+        // mount.nfs might be in */sbin but that ins't always in
+        // $PATH, such as when connecting from CentOS to another
+        // machine via SSH as non-root.
+        let std_path = "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin";
+        return cockpit.spawn([ "which", "mount.nfs" ], { err: "message", environ: [ std_path ] }).then(
```

Comment:
That  feels rather ad-hoc. So the root cause is that the default CentOS CI $PATH does not include /sbin and friends for users? (Which makes sense). We already set a default `$PATH` in cockpit-bridge, but that wouldn't apply if a $PATH is already set -- which is usually the case through ssh.

Since the NFS functionality already needs root anyway (`superuser: "try"` for `nfs_mounts_py`), it would seem fair to use the same option for this `which` call. This should fix the `$PATH` issue, and as well hide the NFS functionality for non-admin users, which seems desirable as well?

Reply:
> Since the NFS functionality already needs root anyway (`superuser: "try"` for `nfs_mounts_py`), it would seem fair to use the same option for this `which` call. This should fix the `$PATH` issue, and as well hide the NFS functionality for non-admin users, which seems desirable as well?

Feature detection should work even if the user is not able to actually use it.  We don't want to hide the NFS panel for non-wheel users, we want to show it with a disabled button.  And in fact, if a non-wheel user is logged in directly, the check works.


So, we really want to check whether the nfs-utils package is installed (nfs-common on the Debians), and the most portable and efficient way seemed to be to just check whether the mount.nfs binary is there.

We could check a specific path directly like `test -f /usr/sbin/mount.nfs` and then make that build-time configurable.

Or we check `test -f /usr/sbin/mount.nfs || test -f /sbin/mount.nfs` to cover the two cases that will happen in practice.

Or we let `which` do the checking with a hard coded $PATH, which is the same as the previous paragraph.

That was my thinking here, anyway...

Reply:
Thanks for the details. I wasn't aware that our NFS page worked without root.